eta/core/datasets.py

```diff
@@ -759,6 +759,11 @@ def add_file(self, data_path, labels_path, new_data_filename=None,
         new_labels_path = os.path.join(labels_subdir, new_labels_filename)
         if labels_path != new_labels_path:
             labels_method(labels_path, new_labels_path)
+            # renames the 'filename' attribute within the json file to reflect
+            # the updated data_filename
+            data = load_json(new_labels_path)
+            data['filename'] = os.path.basename(new_data_path)
+            write_json(data, new_labels_path)
```

Comment:
Better would be to use `self._read_labels()` instead of `load_json()`, because the former would read an actual `ImageLabels` or `VideoLabels` object. So you could do
```python
labels = self._read_labels(new_labels_path)
labels.filename = os.path.basename(new_data_path)
labels.write_json(new_labels_path)
```

Reply:
good catch. yeah it's likely a red flag to even be importing `load_json` and `write_json` from `eta.core.serial`